metrics.go

```diff
@@ -1,60 +1,59 @@
 package logr
 
-import (
-	"fmt"
-	"sync/atomic"
+import "errors"
+
+const (
+	DefMetricsUpdateFreqMillis = 15000 // 15 seconds
 )
 
-// SizeAndCap represents the metrics of something that has a current size
-// and capacity such as a channel or buffer.
-type SizeAndCap struct {
-	Size int
-	Cap  int
+type Counter interface {
+	// Inc increments the counter by 1. Use Add to increment it by arbitrary non-negative values.
+	Inc()
+	// Add adds the given value to the counter. It panics if the value is < 0.
+	Add(float64)
 }
 
-// TargetMetrics provides a snapshot of a target's metrics such as
-// current queue size and capacity.
-type TargetMetrics struct {
-	Queue        SizeAndCap
-	LoggedCount  uint64
-	ErrorCount   uint64
-	DroppedCount uint64
-	BlockedCount uint64
+type Gauge interface {
+	// Set sets the Gauge to an arbitrary value.
+	Set(float64)
+	// Add adds the given value to the Gauge. (The value can be negative, resulting in a decrease of the Gauge.)
+	Add(float64)
+	// Sub subtracts the given value from the Gauge. (The value can be negative, resulting in an increase of the Gauge.)
+	Sub(float64)
 }
 
-// TargetWithMetrics is a target that provides metrics.
-type TargetWithMetrics interface {
-	GetMetrics() TargetMetrics
+type MetricsCollector interface {
+	// QueueSizeGauge returns a Gauge that will be updated by the named target.
```

Comment:
What do you think of include in the godoc what are those field aiming for?

Reply:
Comments added via https://github.com/mattermost/logr/pull/2/commits/da17395268f6ff1c9d105225a30305b02271cae9